src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -317,29 +292,23 @@ static ucs_status_t uct_rdmacm_cm_create_dummy_qp(struct rdma_cm_id *id,
 static ucs_status_t uct_rdamcm_cm_ep_create_qpn(uct_rdmacm_cm_ep_t *cep,
                                                 uint32_t* qpn)
 {
-    struct ibv_cq *cq;
+    struct uct_rdmacm_cm_qp_context *ctx;
     ucs_status_t status;
-    bool supported;
 
-    status = uct_rdamcm_cm_ep_is_reserved_qpn_supported(cep, &supported);
-    if (status != UCS_OK) {
-        return status;
-    }
+    status = uct_rdmacm_cm_get_qp_context(uct_rdmacm_cm_ep_get_cm(cep),
+                                          cep->id->verbs, &ctx);
+    ucs_assert(status == UCS_OK);
 
-    if (supported) {
-        status = uct_rdamcm_cm_ep_create_reserved_qpn(cep);
+    if (ctx->reserved_qpn_support) {
+        status = uct_rdamcm_cm_ep_create_reserved_qpn(cep, ctx);
         if (status == UCS_OK) {
             cep->flags |= UCT_RDMACM_CM_EP_QPN_CREATED;
             *qpn = cep->qpn;
         }        
```

Comment:
common code with `else` branch below

Reply:
@evgeny-leksikov No fully common. In else branch, it is *qpn = cep->**qp->qp_num**;
The common part is cep->flags |= UCT_RDMACM_CM_EP_QPN_CREATED;, but need to add another condition checking to abstract it out.